pgmpy/base/AncestralBase.py

```diff
@@ -0,0 +1,152 @@
+from collections import deque
+from typing import Hashable, Iterable, Optional
+
+import networkx as nx
+import numpy as np
+
+
+class AncestralBase(nx.DiGraph):
+    def __init__(
+        self,
+        ebunch: Optional[Iterable[tuple[Hashable, Hashable]]] = None,
+        latents: set[Hashable] = set(),
+    ):
+        super().__init__()
+        if ebunch:
+            self.add_edges_from(ebunch)
+        self.latents = set(latents)
+
+    def to_adjacency_matrix(self):
+        """
+        Return adjacency matrix with edge marks and node-to-index mapping.
+
+        Returns
+        -------
+        M : np.ndarray
+            A square matrix of shape (n_nodes, n_nodes) where M[i, j]
+            is the mark at node j for edge (i, j).
+        node_index : dict
+            Mapping from node label to row/col index.
+        """
+        nodes = list(self.nodes)
+        n = len(nodes)
+        node_index = {node: i for i, node in enumerate(nodes)}
+
+        M = np.full((n, n), "", dtype=object)
```

Comment:
Having the default as `0`  instead of `""` is better as it is a default convention used everywhere.